courses/api.py

```diff
@@ -579,3 +584,106 @@ def is_program_text_id(item_text_id):
         bool: True if the given id is a program id
     """
     return item_text_id.startswith(PROGRAM_TEXT_ID_PREFIX)
+
+
+def process_course_run_grade_certificate(course_run_grade):
+    """
+    Ensure that the course run certificate is in line with the values in the course run grade
+
+    Args:
+        course_run_grade (courses.models.CourseRunGrade): The course run grade for which to generate/delete the certificate
+
+    Returns:
+        Tuple[ CourseRunCertificate, bool, bool ]: A Tuple containing None or CourseRunCertificate object,
+            A bool representing if the certificate is created, A bool representing if a certificate is deleted
+    """
+    user = course_run_grade.user
+    course_run = course_run_grade.course_run
+
+    # A grade of 0.0 indicates that the certificate should be deleted
+    should_delete = not bool(course_run_grade.grade)
+    should_create = course_run_grade.is_certificate_eligible
+
+    if should_delete:
+        delete_count, _ = CourseRunCertificate.objects.filter(
+            user=user, course_run=course_run
+        ).delete()
+        return None, False, (delete_count > 0)
+
+    elif should_create:
+        certificate, created = CourseRunCertificate.objects.get_or_create(
+            user=user, course_run=course_run
+        )
+        return certificate, created, False
+    return None, False, False
+
+
+def get_certificate_grade_eligible_runs():
+    """
+    Get the list of course runs that are eligible for Grades update/creation and certificates creation
+    """
+    now = now_in_utc()
+    # Get all the course runs valid course runs for certificates/Grades
+    # For a valid run it would be live,
+    # .. end_date would be in future with addition of delay settings.CERTIFICATE_CREATION_DELAY_IN_HOURS
+
+    course_runs = CourseRun.objects.live().filter(
+        Q(end_date__isnull=True)
+        | Q(
+            end_date__gt=now
+            - timedelta(hours=settings.CERTIFICATE_CREATION_DELAY_IN_HOURS)
+        )
+    )
+    return course_runs
+
+
+def generate_course_run_certificates():
+    """
+    Hits the edX grades API for eligible course runs and generates the certificates and grades for users for course runs
+    """
+    now = now_in_utc()
+    course_runs = get_certificate_grade_eligible_runs()
```

Comment:
I think you should be passing `now` to this so that the logic across this function uses the same value (this is realistically probably not an issue, but consistency is probably good).

Other than that, LGTM so feel free to merge after changing that.